src/controllers/genresController.ts

```diff
@@ -0,0 +1,13 @@
+import { GenreModel } from '../models/genreModel.ts';
+
+import * as handlerFactory from '../utils/handlerFactory.ts';
+
+export const getAllGenres = handlerFactory.getAll(GenreModel);
```

Comment:
Could you please explain the advantages of factory usage?
Factory - is a creation design pattern, it solves the problem of creation object without specifying direct class, based on some conditions.  Here you are able to unify some methods but if requirements changed you would need to modify single method which increases the risk of side effects

Reply:
What would you do if one of the routes needed to receive data from two or more sources?

Reply:
handlerFactory abstracts basic CRUD operations and only them. It's beneficial because it follows the DRY principle and forces consistency of CRUD operations throughout all the routes where it is used. But it is in no way limiting, if I need a different functionality - I can implement it as I do in moviesController, for example, where I use factory for CRUD and implement a new function to get movies by genre from scratch.